src/core/linalg/src/sparse/4C_linalg_map.cpp

```diff
@@ -66,6 +66,31 @@ std::unique_ptr<const Core::LinAlg::Map> Core::LinAlg::Map::create_view(const Ep
   ret->map_ = Utils::make_view<Epetra_Map>(const_cast<Epetra_Map*>(&view));
   return ret;
 }
+std::unique_ptr<Core::LinAlg::Map> Core::LinAlg::Map::create_view(Epetra_BlockMap& view)
+{
+  // cast the Epetra_BlockMap to Epetra_Map, this cast can go wrong.
+  const Epetra_Map* epetra_map_ptr = static_cast<const Epetra_Map*>(&view);
+
+  // We may safely cast away const here, since the returned object is const and we use a
+  // OwnerOrView to ensure that we only ever access the viewed map through
+  // const methods.
+  std::unique_ptr<Map> ret(new Map);
+  ret->map_ = Utils::make_view<Epetra_Map>(const_cast<Epetra_Map*>(epetra_map_ptr));
```

Comment:
Also, for the non-const version, just don't cast to const Epetra_Map above, then you don't need const_cast here.